src/main/java/com/google/api/codegen/discovery/transformer/php/PhpSampleTypeNameConverter.java

```diff
@@ -0,0 +1,105 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.transformer.php;
+
+import com.google.api.codegen.discovery.config.TypeInfo;
+import com.google.api.codegen.discovery.transformer.SampleTypeNameConverter;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypedValue;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Field;
+
+public class PhpSampleTypeNameConverter implements SampleTypeNameConverter {
+
+  /** A map from primitive types in proto to PHP counterparts. */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_TYPE_MAP =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_BOOL, "Boolean")
+          .put(Field.Kind.TYPE_INT32, "Integer")
+          .put(Field.Kind.TYPE_INT64, "String")
+          .put(Field.Kind.TYPE_UINT32, "Integer")
+          .put(Field.Kind.TYPE_UINT64, "String")
+          .put(Field.Kind.TYPE_FLOAT, "Float")
+          .put(Field.Kind.TYPE_DOUBLE, "Float")
+          .put(Field.Kind.TYPE_STRING, "String")
+          .put(Field.Kind.TYPE_ENUM, "String")
+          .build();
+
+  /** A map from primitive types in proto to zero value in PHP */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_ZERO_VALUE =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_BOOL, "false")
+          .put(Field.Kind.TYPE_INT32, "0")
+          .put(Field.Kind.TYPE_INT64, "'0'")
+          .put(Field.Kind.TYPE_UINT32, "0")
+          .put(Field.Kind.TYPE_UINT64, "'0'")
+          .put(Field.Kind.TYPE_FLOAT, "0.0")
+          .put(Field.Kind.TYPE_DOUBLE, "0.0")
+          .put(Field.Kind.TYPE_STRING, "''")
+          .put(Field.Kind.TYPE_ENUM, "''")
+          .build();
+
+  private final String apiTypeName;
+
+  public PhpSampleTypeNameConverter(String apiTypeName) {
+    this.apiTypeName = apiTypeName;
+  }
+
+  @Override
+  public TypeName getServiceTypeName(String apiTypeName) {
+    return new TypeName(apiTypeName);
+  }
+
+  @Override
+  public TypeName getRequestTypeName(String apiTypeName, TypeInfo typeInfo) {
+    // N/A
+    return null;
+  }
+
+  @Override
+  public TypeName getTypeName(TypeInfo typeInfo) {
+    // This method is only ever called for the request body type, which should
+    // always be a message.
+    if (typeInfo.isMessage()) {
+      return new TypeName(apiTypeName + "_" + typeInfo.message().typeName());
+    }
+    return getTypeNameForElementType(typeInfo);
```

Comment:
Oh, I've just been doing it to save writing the exception out twice. In other languages `getTypeName` does call `getTypeNameForElementType`, so it seemed natural. In the unexpected case that the transformer does happen to call the element function it could be useful to get an stacktrace instead of null in snippet